src/actions.py

```diff
@@ -79,6 +79,8 @@ def view_back_action(_action, _target, self):
     if self.canvas.toolbar.back_enabled:
         self.canvas.toolbar.back()
         self.canvas.apply_limits()
+        # Fix weird view on back
+        self.canvas.draw()
```

Comment:
````suggestion
graphs.refresh(self)
```

Actually `self.canvas.draw()` also seems to get stuck. The only thing that actually seems consistent so far is a full refresh. Which still feels as snappy as without the fix. So I think that should work.

Reply:
Actually `self.canvas.draw()` also seems to get stuck. The only thing that actually seems consistent so far is a full refresh. Which still feels as snappy as without the fix. So I think that should work.